tests/unit-tests/test_gnn.py

```diff
@@ -2471,6 +2512,448 @@ def check_call_gen_embs(skip_last_self_loop):
+                                      node_feats='feat',
+                                      train_task=False)
+    for i, (input_nodes, seeds, blocks) in enumerate(dataloader4):
+        nfeat_fields = dataloader4.node_feat_fields
+        ninput_feats = np_data.get_node_feats(input_nodes, nfeat_fields)
+        efeat_fields = dataloader4.edge_feat_fields
+        einput_feats_list = np_data.get_blocks_edge_feats(blocks, efeat_fields)
+
+        lbl = np_data.get_node_feats(seeds, 'label')
+
+        with assert_raises(AssertionError):
+            model4(blocks, ninput_feats, einput_feats_list, lbl, input_nodes)
+
+    th.distributed.destroy_process_group()
+    dgl.distributed.kvstore.close_kvstore()
+
+def test_rgcn_edge_model_forward():
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        ep_data = GSgnnData(part_config=part_config,
+                            node_feat_field="feat",
+                            edge_feat_field="feat")
+
+    # Test case 0: normal case, edge model uses edge feature, and provides edge feature
+    #              for mini-batch forward
+    edge_feat_name = {('n0', 'r0', 'n1'): 'feat', ('n0', 'r1', 'n1'): 'feat'}
+    model1 = create_rgcn_edge_model(ep_data.g, num_ffn_layers=0, edge_feat_name=edge_feat_name)
+    target_idx = {("n0", "r1", "n1"): [0, 1]}
+    dataloader1 = GSgnnEdgeDataLoader(ep_data,
+                                      target_idx,
+                                      fanout=[-1, -1],
+                                      batch_size=10,
+                                      label_field='label',
+                                      node_feats='feat',
+                                      edge_feats='feat',
+                                      train_task=False,
+                                      remove_target_edge_type=False)
+    for i, (input_nodes, batch_graph, blocks) in enumerate(dataloader1):
+        nfeat_fields = dataloader1.node_feat_fields
+        ninput_feats = ep_data.get_node_feats(input_nodes, nfeat_fields)
+        efeat_fields = dataloader1.edge_feat_fields
+        einput_feats_list = ep_data.get_blocks_edge_feats(blocks, efeat_fields)
+
+        label_field = dataloader1.label_field
+        lbl = ep_data.get_edge_feats({("n0", "r1", "n1"): [0, 1]}, label_field)
+        loss1 = model1(blocks, batch_graph, ninput_feats, einput_feats_list,
+                       None, lbl, input_nodes)
+        assert th.is_floating_point(loss1)
+
+    # Test case 1: normal case, edge model dose not use edge feature, and does not 
```

Comment:
ditto

Reply:
Add another test case, this will trigger an assert error. In addition, this case has been tested in test_embed.py line https://github.com/awslabs/graphstorm/pull/281.